shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/ClangArgs.kt

```diff
@@ -72,19 +84,16 @@ class ClangArgs(private val configurables: Configurables) : Configurables by con
                 KonanTarget.IOS_X64 ->
                     listOf("-stdlib=libc++", "-isysroot", absoluteTargetSysRoot, "-miphoneos-version-min=9.0.0")
 
-                KonanTarget.ANDROID_ARM32 ->
-                    listOf("-target", targetArg!!,
-                            "--sysroot=$absoluteTargetSysRoot",
-                            // HACKS!
-                            "-I$absoluteTargetSysRoot/usr/include/c++/4.9.x",
-                            "-I$absoluteTargetSysRoot/usr/include/c++/4.9.x/arm-linux-androideabi")
-
-                KonanTarget.ANDROID_ARM64 ->
-                    listOf("-target", targetArg!!,
-                            "--sysroot=$absoluteTargetSysRoot",
-                            // HACKS!
-                            "-I$absoluteTargetSysRoot/usr/include/c++/4.9.x",
-                            "-I$absoluteTargetSysRoot/usr/include/c++/4.9.x/aarch64-linux-android")
+                KonanTarget.ANDROID_ARM32, KonanTarget.ANDROID_ARM64, KonanTarget.ANDROID_X64 -> {
+                    val clangTarget = targetArg!!
+                    val architectureDir = Android.architectureForTarget(target)
+                    val toolchainSysroot = "$absoluteTargetToolchain/sysroot"
+                    listOf("-target", clangTarget,
+                            "--sysroot=$absoluteTargetSysRoot/android-${Android.API}/arch-$architectureDir",
+                            "-I$toolchainSysroot/usr/include/c++/v1",
+                            "-I$toolchainSysroot/usr/include",
+                            "-I$toolchainSysroot/usr/include/$clangTarget")
+                }
```

Comment:
Is this equal to what Android Studio uses to compile C++ sources with NDK?

Reply:
Similar.